src/gsy_e/models/myco_matcher/myco_internal_matcher.py

```diff
@@ -59,31 +65,42 @@ def get_matching_algorithm():
 
+    @staticmethod
+    def _match_recommendations(area_uuid: str, area_data: Dict, markets: List,
+                               get_matches_recommendations: Callable) -> None:
+        """Request trade recommendations and match them in the relevant market."""
+        for market in markets:
+            if not market:
+                continue
+            while True:
+                # Perform matching until all recommendations and their residuals are handled.
+                orders = market.orders_per_slot()
+
+                # Format should be: {area_uuid: {time_slot: {"bids": [], "offers": [], ...}}}
+                data = {
+                    area_uuid: {
+                        time_slot: {**orders_data, "current_time": area_data["current_time"]}
+                        for time_slot, orders_data in orders.items()}}
+                bid_offer_pairs = get_matches_recommendations(data)
+                if not bid_offer_pairs:
+                    break
+                trades_occurred = market.match_recommendations(bid_offer_pairs)
+                if not trades_occurred:
+                    break
+
     def match_recommendations(self, **kwargs):
         """Request trade recommendations and match them in the relevant market."""
         for area_uuid, area_data in self.area_uuid_markets_mapping.items():
             markets = [*area_data["markets"], *area_data["settlement_markets"]]
             if global_objects.future_market_counter.is_time_for_clearing(
                     area_data["current_time"]):
                 markets.append(area_data["future_markets"])
-            for market in markets:
-                if not market:
-                    continue
-                while True:
-                    # Perform matching until all recommendations and their residuals are handled.
-                    orders = market.orders_per_slot()
-
-                    # Format should be: {area_uuid: {time_slot: {"bids": [], "offers": [], ...}}}
-                    data = {
-                        area_uuid: {
-                            time_slot: {**orders_data, "current_time": area_data["current_time"]}
-                            for time_slot, orders_data in orders.items()}}
-                    bid_offer_pairs = self._get_matches_recommendations(data)
-                    if not bid_offer_pairs:
-                        break
-                    trades_occurred = market.match_recommendations(bid_offer_pairs)
-                    if not trades_occurred:
-                        break
+            self._match_recommendations(area_uuid, area_data, markets,
+                                        self._get_matches_recommendations)
+
+            if DayAheadMarketCounter.is_time_for_clearing(area_data["current_time"]):
+                self._match_recommendations(area_uuid, area_data, [area_data["day_ahead_markets"]],
+                                            self._get_matches_recommendations_day_ahead)
```

Comment:
If we remove `self.match_algorithm_day_ahead` and `self._get_matches_recommendations_day_ahead` as I mentioned in my previous comment, we could just use the same procedure of the future markets.

Reply:
Even better! Will do.